File: python/283_Move-Zeros.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        #Time:O(n), Space:O(1)
        if (len(nums)==1):
            return

        ptr1, ptr2 = 0, 0
        while (ptr1<len(nums)):
            if nums[ptr1]==0:
                pass
            else:
                nums[ptr2] = nums[ptr1]
                ptr2+=1
            ptr1+=1
        
        for idx in range(ptr2, len(nums)):
            nums[idx]=0

File: c++/283_Move-Zeros.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();

        int z = 0, nz = 0;
        while(nz < n){
            
            while(nz < n && nums[nz] == 0) ++nz;
            while(z < n && nums[z] != 0) ++z;

            if(nz < n && z < nz) swap(nums[z], nums[nz]);
            ++nz;
        }
    }
};